### PR TITLE
🐛 Update user locale on change

### PR DESCRIPTION
**Pull request by mckingho**

### CODE REVIEW

**Comment by mckingho** [comment]
https://github.com/likecoin/likecoin-api-public/pull/24

File: store/modules/actions/user.js [hunk 1]
```diff
@@ -159,6 +159,9 @@ export async function refreshUser({ commit, state, dispatch }) {
       commit(types.UI_INFO_MSG, '');
       commit(types.MISSION_CLEAR_ALL);
     }
+    if (user && user.locale) {
+      commit(types.UI_SET_LOCALE, user.locale);
```
Comment (williamchong007) [inline store/modules/actions/user.js]: use `dispatch`

File: store/modules/actions/user.js [hunk 2]
```diff
@@ -171,6 +174,9 @@ export async function refreshUserInfo({ commit, dispatch }) {
   if (user) {
     commit(types.USER_SET_USER_INFO, user);
   }
+  if (user && user.locale) {
+    commit(types.UI_SET_LOCALE, user.locale);
```
Comment (williamchong007) [inline store/modules/actions/user.js]: use `dispatch`

File: components/LanguageSwitch.vue [hunk 3]
```diff
@@ -63,13 +63,22 @@ export default {
...
   },
+  computed: {
+    ...mapGetters([
+      'getCurrentLocale',
+    ]),
+  },
+  watch: {
+    getCurrentLocale(locale) {
+      this.$i18n.loadLanguageAsync(locale);
+      this.$cookie.set('language', locale, { expires: '1M', secure: true });
```
Comment (williamchong007) [inline components/LanguageSwitch.vue]: do this in `setLocale` instead of a component

**Pull request merged**